### PR TITLE
Add models

### PR DESCRIPTION
**Pull request by dinoocch**
This commit adds model functionality, as well as tests for each of the
features added.  In particular, I choose to use NoSQL elements in order
to speed up execution and make implementation easier.  Since we do not
need JOINS on element or submission information, the disadvantages of
NoSQL are greatly reduced.

In addition, this commit refactors the tests and continuous integration
to use Postgresql and configures flake8 to ignore automatically
generated migrations.

### CODE REVIEW

**Comment by dinoocch** [comment]
## Coverage
```
(env) dino Forms [models-real] % ../env/bin/coverage report --include \*/models/\*
Name                                                     Stmts   Miss  Cover
----------------------------------------------------------------------------
constellation_forms/models/__init__.py                       3      0   100%
constellation_forms/models/form.py                          29      0   100%
constellation_forms/models/formSubmission.py                37      0   100%
constellation_forms/models/validator.py                     15      0   100%
constellation_forms/tests/models/__init__.py                 0      0   100%
constellation_forms/tests/models/testForm.py                26      0   100%
constellation_forms/tests/models/testFormSubmission.py     112      0   100%
constellation_forms/tests/models/testValidator.py           13      0   100%
----------------------------------------------------------------------------
TOTAL                                                      235      0   100%
```

**Comment by the-maldridge** [comment]
In the validator slug you use the widget slug token instead of a specific one for the validator please change this before merging

**Comment by dinoocch** [comment]
Whole docstring on that model was copy and pasted from widget.py and I forgot to update it. Fixed.

**Comment by the-maldridge** [comment]
Ok. I'll put in a"review" when I'm back at my laptop so the merge will go
green. @m_wynn, you have final approval on this merge.

File: constellation_forms/models/__init__.py
Comment (m-wynn) [inline constellation_forms/models/__init__.py]: that can be deleted?  label?

File: constellation_forms/models/formSubmission.py
Comment (m-wynn) [inline constellation_forms/models/formSubmission.py]: These should be a different error message

File: constellation_forms/tests/models/testFormSubmission.py
Comment (m-wynn) [inline constellation_forms/tests/models/testFormSubmission.py]: This could be cleaner if you put a lot of the duplicate code into a single function.  Thoughts @the-maldridge ?

**Review by m-wynn — CHANGES_REQUESTED** [review]

File: constellation_forms/models/__init__.py
Comment (dinoocch) [reply constellation_forms/models/__init__.py]: Deleted

File: constellation_forms/models/formSubmission.py
Comment (dinoocch) [reply constellation_forms/models/formSubmission.py]: Fixed.

File: constellation_forms/tests/models/testFormSubmission.py
Comment (dinoocch) [reply constellation_forms/tests/models/testFormSubmission.py]: Code reuse reduced...

**Review by m-wynn — APPROVED** [review]

**Review by the-maldridge — APPROVED** [review]
All looks good except for the previously mentioned settings file comments.

**Pull request merged**
